src/number_theory/cyclotomic/basic.lean

```diff
@@ -104,17 +104,30 @@ lemma singleton_one [h : is_cyclotomic_extension {1} A B] : (⊥ : subalgebra A
 algebra.eq_top_iff.2 (λ x, by simpa [adjoin_singleton_one]
   using ((is_cyclotomic_extension_iff _ _ _).1 h).2 x)
 
+variables {A B}
+
+/-- If `(⊥ : subalgebra A B) = ⊤`, then `is_cyclotomic_extension ∅ A B`. -/
+lemma singleton_zero_of_bot_eq_top (h : (⊥ : subalgebra A B) = ⊤) :
+  is_cyclotomic_extension ∅ A B :=
+begin
+  refine (iff_adjoin_eq_top _ _ _).2 ⟨λ s hs, by simpa using hs, _root_.eq_top_iff.2 (λ x hx, _)⟩,
+  rw [← h] at hx,
+  simpa using hx,
+end
+
+variables (A B)
+
 /-- Transitivity of cyclotomic extensions. -/
-lemma trans (C : Type w) [comm_ring C] [nontrivial C] [algebra A C] [algebra B C]
+lemma trans (C : Type w) [comm_ring C] [algebra A C] [algebra B C]
   [is_scalar_tower A B C] [hS : is_cyclotomic_extension S A B]
-  [hT : is_cyclotomic_extension T B C] [no_zero_smul_divisors B C] :
+  [hT : is_cyclotomic_extension T B C] (h : function.injective (algebra_map B C)) :
```

Comment:
these two are equivalent, iirc; why are we preferring having more arguments lying around?

Reply:
They're not unfortunately. They are equivalent if `C` is a domain, or something like that, the implication `no_zero_smul_divisors → injective` is always true though, so this is strictly more general. But I can add a primed version to use `no_zero_smul_divisors`.

Reply:
oh, i see :((

Reply:
I don't think having a primed vesrion is needed